jormungandr/src/metrics/backends/simple_counter.rs

```diff
@@ -101,45 +101,46 @@ impl Default for SimpleCounter {
 
 impl MetricsBackend for SimpleCounter {
     fn add_tx_recv_cnt(&self, count: usize) {
-        self.tx_recv_cnt.fetch_add(count, Ordering::SeqCst);
+        self.tx_recv_cnt.fetch_add(count, Ordering::Relaxed);
     }
 
     fn set_tx_pending_cnt(&self, count: usize) {
-        self.tx_pending_cnt.store(count, Ordering::SeqCst);
+        self.tx_pending_cnt.store(count, Ordering::Relaxed);
     }
 
     fn add_block_recv_cnt(&self, count: usize) {
-        self.block_recv_cnt.fetch_add(count, Ordering::SeqCst);
+        self.block_recv_cnt.fetch_add(count, Ordering::Relaxed);
     }
 
     fn add_peer_connected_cnt(&self, count: usize) {
-        self.peers_connected_cnt.fetch_add(count, Ordering::SeqCst);
+        self.peers_connected_cnt.fetch_add(count, Ordering::Relaxed);
     }
 
     fn sub_peer_connected_cnt(&self, count: usize) {
-        self.peers_connected_cnt.fetch_sub(count, Ordering::SeqCst);
+        self.peers_connected_cnt.fetch_sub(count, Ordering::Relaxed);
     }
 
     fn add_peer_quarantined_cnt(&self, count: usize) {
         self.peers_quarantined_cnt
-            .fetch_add(count, Ordering::SeqCst);
+            .fetch_add(count, Ordering::Relaxed);
     }
 
     fn sub_peer_quarantined_cnt(&self, count: usize) {
         self.peers_quarantined_cnt
-            .fetch_sub(count, Ordering::SeqCst);
+            .fetch_sub(count, Ordering::Relaxed);
     }
 
     fn add_peer_available_cnt(&self, count: usize) {
-        self.peers_available_cnt.fetch_add(count, Ordering::SeqCst);
+        self.peers_available_cnt.fetch_add(count, Ordering::Relaxed);
     }
 
     fn sub_peer_available_cnt(&self, count: usize) {
-        self.peers_available_cnt.fetch_sub(count, Ordering::SeqCst);
+        self.peers_available_cnt.fetch_sub(count, Ordering::Relaxed);
```

Comment:
I think this could potentially overflow and wrap around, if the `fetch_sub` gets executed before the `fetch_add`, although I guess it would be a strange cornercase...

I'm not that fresh on memory ordering details, though.

Reply:
Well, I think that is technically possible even with `SeqCst`/`AcqRel` if we somehow manage to try to remove a peer that is not yet counted, there should be more sophisticated synchronization. I think, the Prometheus integration works under the same assumption since under the hood their counters are just atomics with all operations done with the `Relaxed` ordering.

Reply:
These operations are documented to wrap around, but the ordering needs to be `AcqRel` I think.